src/App/Fossa/VSI/Fingerprint.hs

```diff
@@ -0,0 +1,181 @@
+sinkHash = sink hashInit
+  where
+    sink ctx = do
+      b <- await
+      case b of
+        Nothing -> return $! hashFinalize ctx
+        Just bs -> sink $! hashUpdate ctx bs
+
+-- | Hashes the whole contents of the given file in constant memory.
+-- Adapted from @hashFile@ in https://hackage.haskell.org/package/cryptonite-conduit-0.2.2/docs/src/Crypto-Hash-Conduit.html
+hashFile :: (Has (Lift IO) sig m, Has Diagnostics sig m, HashAlgorithm hash) => FilePath -> m (Digest hash)
+hashFile fp =
+  sendIO (runConduitRes (sourceFile fp .| sinkHash))
+    `catch` (\(e :: IOException) -> fatalText ("unable to hash file: " <> toText (show e)))
+
+hashFileCommentStripped :: (Has (Lift IO) sig m, Has Diagnostics sig m, HashAlgorithm hash) => FilePath -> m (Digest hash)
+hashFileCommentStripped file =
+  sendIO
+    ( runConduitRes $
+        sourceFile file -- Read from the file
+          .| decodeUtf8C -- Decode to text
+          .| basicCStyleCommentStripC -- Strip comments
+          .| encodeUtf8C -- Encode back to bytes
+          .| sinkHash -- Hash the result
+    )
+    `catch` (\(e :: IOException) -> fatalText ("unable to hash file: " <> toText (show e)))
+
+fingerprintRaw :: (Has (Lift IO) sig m, Has Diagnostics sig m) => Path Abs File -> m (Fingerprint Raw)
+fingerprintRaw file = do
+  (fp :: Digest SHA256) <- hashFile $ toFilePath file
+  pure $ encodeFingerprint fp
+
+fingerprintContentsRaw :: (Has ReadFS sig m, Has Diagnostics sig m, Has (Lift IO) sig m) => Path Abs Dir -> m [Fingerprint Raw]
+fingerprintContentsRaw = walk' $ \_ _ files -> do
+  fps <- traverse fingerprintRaw files
+  pure (fps, WalkContinue)
+
+fingerprintCommentStripped :: (Has ReadFS sig m, Has (Lift IO) sig m, Has Diagnostics sig m) => Path Abs File -> m (Maybe (Fingerprint CommentStripped))
+fingerprintCommentStripped file = contentIsBinary file >>= doFingerprint
+  where
+    doFingerprint True = pure Nothing -- Don't attempt to comment strip binary files
+    doFingerprint False = do
+      (fp :: Digest SHA256) <- hashFileCommentStripped $ toFilePath file
+      pure . Just $ encodeFingerprint fp
+
+fingerprint :: (Has ReadFS sig m, Has (Lift IO) sig m, Has Diagnostics sig m) => Path Abs File -> m Combined
+fingerprint file =
+  Combined
+    <$> fingerprintRaw file
+    <*> fingerprintCommentStripped file
+
+-- | This implementation is based on the comment strip logic from the internal logic used when crawling OSS components.
+-- It is very basic:
+--
+-- * Only works for C-style comments
+-- * Only catches @\\n@ newlines
+-- * Doesn't handle edge cases (escaped comments for example)
+-- * Also omits any blank lines
+--
+-- Despite these drawbacks, we have to reimplement it the same way so that fingerprints line up correctly in the VSI analysis service.
+basicCStyleCommentStripC :: Monad m => ConduitT Text Text m ()
+basicCStyleCommentStripC = splitOnUnboundedE (== '\n') .| process .| mapC Text.strip .| filterC (not . Text.null) .| bufferedNewline Nothing
```

Comment:
Looks like `linesUnboundedC` is implemented as `splitOnUnboundedE (== '\n')`.  So it's not causing the windows issues.  We should keep this as `linesUnboundedC`.

https://hackage.haskell.org/package/conduit-1.3.4.2/docs/src/Data.Conduit.Combinators.Unqualified.html#linesUnboundedC

which leads to 

https://hackage.haskell.org/package/conduit-1.3.4.2/docs/src/Data.Conduit.Combinators.html#linesUnbounded

Reply:
Good point, probably it's the git checkout thing like you mentioned.

Thinking about it, we should probably just convert any `\r\n` to `\n` to keep it consistent with our indexers (which all ran on Linux). While this'll cause us to not match files that truly contain `\r\n`, it's probably an overall more correct thing to do. I'll make that change.